### PR TITLE
Paginated embed and League of Legends free champion rotation

### LINKED ISSUE #76: Paged embed system

### PR DESCRIPTION
**Pull request by dpaiv0**
+ Closes #76
+ Based in [Klasa](https://klasa.js.org/)'s [RichDisplay](https://fratik.github.io/klasa/tutorial-RichDisplay_.html)

image
image

```js
async run ({ t, channel }, author) {
  channel.startTyping()
  const PaginatedEmbed = new SwitchbladeEmbed.PaginatedEmbed(t)
  PaginatedEmbed
    .addPage(new SwitchbladeEmbed(author).setTitle('owo page one').setDescription('skip to the next one'))
    .addPage(new SwitchbladeEmbed(author).setTitle('goteem').setImage('https://png.pngtree.com/element_origin_min_pic/17/07/31/1751bed6d1682d8ffaedf3a29d5f3981.jpg'))
    .run(await channel.send('owo'))
  channel.stopTyping()
}
```

### CODE REVIEW

File: src/apis/LeagueOfLegends.js [hunk 1]
```diff
@@ -41,12 +43,40 @@ module.exports = class LeagueOfLegends extends APIWrapper {
...
+    })
+  }
+
+  async fetchChampionRotation (language = 'en_US', newPlayer = false) {
+    const payload = await this.request('/lol/platform/v3/champion-rotations', false).then(u => u)
+
+    let champions = []
+    const championIds = newPlayer ? payload.freeChampionIdsForNewPlayers : payload.freeChampionIds
+
+    for (var i in championIds) {
```
Comment (metehus) [inline src/apis/LeagueOfLegends.js]: USE FUCKING ARRAY.MAP

File: src/commands/games/leagueoflegends.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const { Command, SwitchbladeEmbed } = require('../../')
 
-const types = ['champion', 'champ', 'c', 'status', 's']
+const buttons = ['Q', 'W', 'E', 'R']
```
Comment (metehus) [inline src/commands/games/leagueoflegends.js]: would be nice the buttons be a custom emote of a key

File: src/structures/ReactionHandler.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    this.stop()
+  }
+
+  eight () {
+    if (this.display.options.length - 1 < 8 + (this.currentPage * 10)) return
+    this.resolve(8 + (this.currentPage * 10))
+    this.stop()
+  }
+
+  nine () {
```
Comment (metehus) [inline src/structures/ReactionHandler.js]: why the fuck 9 methods with the same code but changed 2 numbers

**Comment by dpaiv0** [comment]
remaking this l8r this month

**Comment by pedrofracassi** [comment]
@dpaiv0 can you make it?

**Pull request closed without merging**